### ISSUE TITLE
Cooler matrix as input: Correction (ICE, KR) of the data

### ISSUE THREAD

**Issue by gtrichard** [dozmorovlab/SpectralTAD#3]
Hello, thanks for the great tool.

I have a question concerning the input matrix, they should be KR or ICE corrected right? Using a cool matrix as input, one can choose to output the correction factor as a supplementary column (`-b` option from `cooler dump`). Is there any plan for SpectralTAD to support such corrected matrices as input? I'm not sure cooler dump supports the output of corrected values directly. Or is the correction performed within SpectralTAD (I guess not)?

Or perhaps uncorrected matrices is what SpectralTAD needs to call TADs correctly?

Thank you.

**Comment by mdozmorov** [comment]
Practically, we showed that raw Hi-C data works well, as well as ICE-normalized, in [Figure 2](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03652-w/figures/2). The KR normalization performed less optimal. SpectralTAD by itself does not do any normalization. 

If using normalized data is the goal, we used [HiCExplorer's hicCorrectMatrix](https://hicexplorer.readthedocs.io/en/latest/content/tools/hicCorrectMatrix.html) function. HiCExplorer supports both .hic and .cool formats (conversions), and export in text format. 

Bottom line - raw Hi-C data is OK to use. But I understand that many may want to apply normalization - ICE is better to use then.

Sorry for some delay replying.

**Comment by gtrichard** [comment]
Thanks for the quick answer.

I have to check if HiCExplorer's correction is directly applied to the matrix values when using cool format. So that `cooler dump` actually exports the corrected values directly.

But I think that `hicConvertFormat` to GInteractions should work, this might be a better option than `cooler dump` since many format are supported by `hicConvertFormat`, including .h5, .cool, etc.

**Comment by mdozmorov** [comment]
I believe, we converted `.cool` to `.h5`, normalized with ICE, exported in Ginteractions, and then imported into R.

**Comment by gtrichard** [comment]
Ok, it makes sense. The vignette is recommending to use `cool dump` to work with cool files, so perhaps hicConvertFormat to GInteractions should be mentionned, as well as ICE vs KR.

Thanks for the input !

**Comment by mdozmorov** [comment]
Good point, adding more info about normalization is on my todo. Please, ask anything else, if needed.
